### PR TITLE
Add version locking to YUM commands

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

**Review by freva — APPROVED** [review]
Add a test where the command is on one of the other forms (`name-ver-rel`)?

**Review by freva — APPROVED** [review]

**Comment by hakonhall** [comment]
Added a few more tests, and extracted YumPackageName into its own file to make the builder more readily available to clients, since the String syntax is so ambiguous.

PTAL

**Review by freva — COMMENTED** [review]
Also please add alternative `install()/remove()/upgrade()` that takes in `YumPackageName...`

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumPackageName.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * YUM package name.
+ *
+ * @author hakonhall
+ */
+public class YumPackageName {
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumPackageName.java]: Done

**Comment by hakonhall** [comment]
> Also please add alternative install()/remove()/upgrade() that takes in YumPackageName

In this case, let me extend YPN to parse all existing packages so the internals can use YPN instead of String.

**Comment by hakonhall** [comment]
PTAL @freva

**Review by freva — APPROVED** [review]

**Pull request merged**